tests/responses/test_search_result.py

```diff
@@ -14,32 +15,46 @@
 
 
 class TestSearchResult:
-    @patch("caselawclient.responses.search_result.api_client")
-    def test_init(self, mock_api_client, valid_search_result_xml):
+    def setup_method(self):
+        self.client = MarklogicApiClient(
+            host="",
+            username="",
+            password="",
+            use_https=False,
+            user_agent="marklogic-api-client-test",
+        )
+
+    def test_init(self, valid_search_result_xml):
         """
         GIVEN a valid search result XML and a mock API client
         WHEN initializing a SearchResult object
         THEN the attributes are set correctly
         """
-        mock_api_client.get_properties_for_search_results.return_value = "<foo/>"
-        mock_api_client.get_last_modified.return_value = "bar"
-
-        node = etree.fromstring(valid_search_result_xml)
-        search_result = SearchResult(node)
-
-        assert search_result.uri == "a/c/2015/20"
-        assert search_result.neutral_citation == "[2015] A 20 (C)"
-        assert search_result.name == "Another made up case name"
-        assert search_result.date == datetime.datetime(2017, 8, 8, 0, 0)
-        assert search_result.court is None
-        assert search_result.matches == (
-            "<p data-path=\"fn:doc('/a/c/2015/20.xml')/*:akomaNtoso\">"
-            "text from the document that matched the search</p>\n"
-        )
-        assert search_result.content_hash == "test_content_hash"
-        assert search_result.transformation_date == "2023-04-09T18:05:45"
-        assert etree.tostring(search_result.metadata.node).decode() == "<foo/>"
-        assert search_result.metadata.last_modified == "bar"
+        with (
+            patch.object(
+                self.client, "get_properties_for_search_results"
+            ) as mock_get_properties_for_search_results,
+            patch.object(self.client, "get_last_modified") as mock_get_last_modified,
+        ):
+            mock_get_properties_for_search_results.return_value = "<foo/>"
+            mock_get_last_modified.return_value = "bar"
+
+            node = etree.fromstring(valid_search_result_xml)
+            search_result = SearchResult(node, self.client)
```

Comment:
This feels suspiciously like SearchResult should be a method on the client.

Reply:
I definitely agree, I've made a note in the list of things to refactor that search (as a whole) is in the wrong place. It's a bit of a chunk to move though and has a lot of test changes involved, so it wants to happen in its own PR.